### ISSUE TITLE
Cannot build neovim with Lua plugins

### ISSUE THREAD

**Issue by hongchangwu** [nix-community/home-manager#1531]
### Issue description

This seems to affect only Lua plugins, regular Vim plugins work fine. Here is a minimal repro:

```nix
{ pkgs, ... }:

{
  programs.neovim = {
    enable = true;
    plugins = [ pkgs.vimPlugins.nvim-lspconfig ];
  };
}
````

```
$ home-manager build
these derivations will be built:
  /nix/store/vdf0zlplwfsaijf92q9ix0pcfqrx401k-neovim-0.4.4.drv
  /nix/store/slr0ldfyq7az831qbl1kd9spim3fqm7k-home-manager-path.drv
  /nix/store/daxrmgnpdp84w29g2bqp5cdw11daxhip-activation-script.drv
  /nix/store/rzidy0smhlvggc28ycgssarc6za5qr5i-home-manager-generation.drv
building '/nix/store/vdf0zlplwfsaijf92q9ix0pcfqrx401k-neovim-0.4.4.drv'...
Generating remote plugin manifest
Error detected while processing /nix/store/g4380vh1cpjyxval4hxx2mq9mjhp1z46-vimplugin-nvim-lspconfig-2020-09-02/share/vim-plugins/nvim-lspconfig/plugin/nvim_lsp.vim:
line   14:
E5105: Error while calling lua chunk: .../home-manager/start/nvim-lspconfig/lua/nvim_lsp/util.lua:11: attempt to index local 'lsp' (a nil value)
remote/host: generated rplugin manifest: /nix/store/jihn0kqw7312vnr0mhjdk804vgy39kd3-neovim-0.4.4/rplugin.vim
Generating rplugin.vim failed!
builder for '/nix/store/vdf0zlplwfsaijf92q9ix0pcfqrx401k-neovim-0.4.4.drv' failed with exit code 1
cannot build derivation '/nix/store/slr0ldfyq7az831qbl1kd9spim3fqm7k-home-manager-path.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/rzidy0smhlvggc28ycgssarc6za5qr5i-home-manager-generation.drv': 1 dependencies couldn't be built
error: build of '/nix/store/rzidy0smhlvggc28ycgssarc6za5qr5i-home-manager-generation.drv' failed
```

#### Maintainer CC

@teto @Gerschtli @rycee

#### Technical details

```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.104-microsoft-standard, Ubuntu, 20.04.1 LTS (Focal Fossa)`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.7`
 - channels(hcwu): `"home-manager-20.09, nixpkgs-20.09beta514.a9226f2b3a5"`
 - nixpkgs: `/home/hcwu/.nix-defexpr/channels/nixpkgs`
```

**Comment by cole-h** [comment]
It might be that these plugins require neovim nightly (I know that nvim-lspconfig does: https://github.com/neovim/nvim-lspconfig#requires-nvim-05-head).

**Comment by Gerschtli** [comment]
I didn't test it, but can you build this plugin with plain `nix-build`? It seems to me, that this issue should be reported to nixpkgs.

**Comment by teto** [comment]
@cole-h  is right, lsp is only available on neovim master (~0.5), which is why the plugin can't find the `lsp` variable in 0.4.4m just override neovim source. There are even neovim overlays out there if you look carefully (one was announced on nixos discourse).

**Comment by hongchangwu** [comment]
Indeed it seems like all the Lua plugins I have tried (`nvim-lspcinfg` and `nvim-treesitter`) require neovim 5.0 nightly. Close this one as this is not a problem of the `neovim` module here.
